src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.service.load.LoadService;
+import bio.terra.service.load.flight.LoadMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+// Populate the files to be loaded from the incoming array
+public class IngestPopulateFileStateFromFlightMapStep extends SkippableStep {
+
+  private final LoadService loadService;
+
+  public IngestPopulateFileStateFromFlightMapStep(
+      LoadService loadService, Predicate<FlightContext> skipCondition) {
+    super(skipCondition);
+    this.loadService = loadService;
+  }
+
+  public IngestPopulateFileStateFromFlightMapStep(LoadService loadService) {
+    this.loadService = loadService;
+  }
+
+  @Override
+  public StepResult doSkippableStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
+    UUID loadId = UUID.fromString(workingMap.get(LoadMapKeys.LOAD_ID, String.class));
```

Comment:
It's a shame this is stored as `String` not `UUID` as all uses appear to convert to `UUID` after extracting it.

Using the newer FlightMap code, where each item is serialized individually (vs now where all parameters are serialized as a single json object) you can actually deserialize a `String` directly as a `UUID` even it was serialized as a `String`.